server/chat-plugins/chatlog.ts

```diff
@@ -678,34 +690,41 @@ export const commands: ChatCommands = {
 	sl: 'searchlogs',
 	searchlog: 'searchlogs',
 	searchlogs(target, room) {
+		if (!room) return this.requiresRoom();
 		target = target.trim();
-		const [search, tarRoom, limit, date] = target.split(',').map(str => str.trim());
+		const args = target.split(',').map(item => item.trim());
 		if (!target) return this.parse('/help searchlogs');
-		if (search.length < 3) return this.errorReply(`Too short of a search query.`);
-		if (!search) return this.errorReply('Specify a query to search the logs for.');
-		let limitString;
-		if (/^[0-9]+$/.test(limit)) {
-			limitString = `--limit-${limit}`;
-		} else if (toID(limit) === 'all') {
-			limitString = `--limit-all`;
-		} else if (!limit) {
-			limitString = ``;
-		} else {
-			return this.errorReply(`Cap must be a number or [all].`);
+		let date = 'all';
+		const searches: string[] = [];
+		let limit = '500';
+		let tarRoom = room.roomid;
+		for (const arg of args) {
+			if (/room:(.+)/.exec(arg)) {
+				const id = /room:(.+)/.exec(arg)![1];
```

Comment:
You definitely never want to run a regex twice. Just save the output in a variable.

In this case, though, why are you even using a regex? Just use `arg.startsWith('room:')` and `arg.slice(5)`.